From 1879f40af742b33ee1e09733854b1e89134b977f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=81lvaro=20Fern=C3=A1ndez=20Rojas?= <noltari@gmail.com>
Date: Sun, 21 Feb 2016 10:53:35 +0100
Subject: [PATCH] hwrng: bcm63xx - fix non device tree compatibility
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Allow building when OF is not enabled as suggested by Florian

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>
Reported-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
---
 drivers/char/hw_random/bcm63xx-rng.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/char/hw_random/bcm63xx-rng.c
+++ b/drivers/char/hw_random/bcm63xx-rng.c
@@ -130,17 +130,19 @@ static int bcm63xx_rng_probe(struct plat
 	return 0;
 }
 
+#ifdef CONFIG_OF
 static const struct of_device_id bcm63xx_rng_of_match[] = {
 	{ .compatible = "brcm,bcm6368-rng", },
 	{},
 };
 MODULE_DEVICE_TABLE(of, bcm63xx_rng_of_match);
+#endif
 
 static struct platform_driver bcm63xx_rng_driver = {
 	.probe		= bcm63xx_rng_probe,
 	.driver		= {
 		.name	= "bcm63xx-rng",
-		.of_match_table = bcm63xx_rng_of_match,
+		.of_match_table = of_match_ptr(bcm63xx_rng_of_match),
 	},
 };
 
